mqtt/mqtt-bridge/src/persist/loader.rs

```diff
@@ -75,10 +74,13 @@ where
                         Some((new_key, _)) if new_key == key => {
                             new_batch.pop_front();
                         }
-                        _ => panic!(
-                            "Invalid MessageLoader state: new_batch is corrupted {:?} for a key {}",
-                            new_batch, key
-                        ),
+                        _ => {
+                            return Poll::Ready(Some(Err(PersistError::Loader {
+                                key: *key,
```

Comment:
Is it worth saving the state if the batch is corrupted? Isn't it better to just discard the current state and restart?

Reply:
that's what we do here, right? I copy data to log it in the error message. Also it causes bridge to restart. With RB we will continue from where we started, and for memory we start from clean slate